src/lib/comms/sol-oic-client.c

```diff
@@ -996,7 +996,7 @@ _resource_request(struct sol_oic_client_request *request,
 
 cbor_error:
     SOL_ERR("Could not encode CBOR representation: %s", cbor_error_string(err));
-    r = -ECOMM;
+    r = -ECANCELED;
```

Comment:
I'm wondering if this error is the best for this case. Maybe `EBADMSG` fits better.